src/main/java/org/broadinstitute/hellbender/tools/pon/coverage/pca/HDF5PCACoveragePoNCreationUtils.java

```diff
@@ -455,7 +454,8 @@ static ReadCountCollection removeColumnsWithExtremeMedianCounts(final ReadCountC
 
         // Determine kept and dropped column sets.
         final Set<String> columnsToKeep = new HashSet<>(readCounts.columnNames().size());
-        final Map<String, Double> columnsToDrop = new LinkedHashMap<>((int) (extremeColumnMedianCountPercentileThreshold * 4.0 / 100.0));
+        final int initialMapSize = ((int) (2. * extremeColumnMedianCountPercentileThreshold / 100.)) * readCounts.columnNames().size();
```

Comment:
in response to a comment by @davidbenjamin.  the original value of 4 was probably incorrect or overkill.  we drop extreme values below the percentile threshold and above 1 - the threshold, so the number of values is roughly twice the threshold.  i'm not sure if this actually makes a difference in performance, though (or if being off by one and making the initial map slightly too small will hurt performance).